Apache-Beam-Jobs.md

```diff
@@ -1,55 +1,85 @@
+* **Continuous**: Operations that are designed to run _indefinitely_ by reacting to updates to the datastore. Here are some examples:
 
-Developer Environment
----------------------
-![screencast of running jobs](https://user-images.githubusercontent.com/5094060/128743997-70cca5f9-0b76-4294-806e-f65f5df5be95.gif)
-1. Sign in as an administrator ([instructions][3])
-2. Navigate to **Admin Page > Roles Tab**
-3. Add the "Release Coordinator" role to the username you are signed in with
-4. Navigate to http://localhost:8181/release-coordinator, then to the **Beam Jobs tab**
-5. Search for your job and then click the **Play button**
+  * Updating the top 10 answers to a lesson every time a new answer is submitted.
+  * Generating notifications for the events that users have subscribed to whenever those events change.
 
-Apache Beam Job Architecture
-============================
+If you're already familiar with Apache Beam or are eager to start writing a new job, jump to the [case studies](#case-studies). Otherwise, you can read the whole page. If you still have questions after reading, take a look at the [Apache Beam Programming Guide][1] for more details.
 
-Jobs are composed of the following components:
--   [`Pipeline`s](#pipelines)
--   [`PValue`s](#pvalues)
--   [`PTransform`s](#ptransforms)
--   [`Runner`s](#runners)
+## Running Apache Beam Jobs
 
-## `Pipeline`s
+These instructions assume you are running a local development server. If you are a release coordinator running these jobs on the production or testing servers, you should already have been granted the "Release Coordinator" role, so you can skip steps 1-3.
 
-`Pipeline`s manage a "DAG"
-([directed acyclic graph](https://en.wikipedia.org/wiki/Directed_acyclic_graph))
-of `PValue`s and the `PTransform`s that compute them. Conceptually, `PValue`s
-are the DAG's nodes and `PTransform`s are the edges.
+1. Sign in as an administrator ([instructions][3]).
+2. Navigate to **Admin Page > Roles Tab**.
+3. Add the "Release Coordinator" role to the username you are signed in with.
+4. Navigate to http://localhost:8181/release-coordinator, then to the **Beam Jobs tab**.
+5. Search for your job and then click the **Play button**.
+6. Click "Start new job".
 
-For example:
+![Screen recording showing how to run jobs](https://user-images.githubusercontent.com/5094060/128743997-70cca5f9-0b76-4294-806e-f65f5df5be95.gif)
+
+## Apache Beam Job Architecture
+
+Conceptually, an Apache Beam job is just a bunch of steps, each of which transforms some input data into some output data. For example, if you wanted to count how many interactions are in all of Oppia's explorations, you could break that task down into a series of transformations:
+
+```text
++--------------+ Count interactions +-----------------------------+ Sum +-------+
+| Explorations | -----------------> | (exploration, count) Tuples | --> | Count |
++--------------+                    +-----------------------------+     +-------+
+```
```

Comment:
```suggestion

    .--------------. Count interactions .--------. Sum .-------.
    | Explorations | -----------------> | Counts | --> | Total |
    '--------------'                    '--------'     '-------'

```

- Reformatted the graph to stay consistent with the others.
- This example doesn't require tuples, since we only care about counts, for example:
- ```python
  exp_pcoll
  | beam.Map(lambda exp: len(exp.interactions))
  | beam.CombineGlobally(sum)
  ```

Reply:
done